### ISSUE TITLE
Component methods uses xpathes from another Components

### ISSUE THREAD

**Issue by DimaGrinevich** [zebrunner/carina#1409]
I am trying to:
```
        List<ResultTestMethodCard> methodCards = page.getTestMethods();
        for (ResultTestMethodCard methodCard : methodCards) {
            if (methodCard.isErrorStacktracePresent()) {
              ...
              some code
              ...
            } else {
              ...
              another code
              ...
            }
        }
```
ResultTestMethodCard extends AbstractUIObject,
xpath for ErrorStacktrace element written with .//

If there at least one visible ErrorStacktrace element out of all method cards, than all methods cards will consider, that their ErrorStacktrace element is also visible

**Comment by tvtester** [comment]
I'm experiencing similar problem with my component class objects which basically extends AbstractUIObject

In my case, if we have more than 1 swimlane object with multiple feedcards, then if specific feed card property (i.e. text1, text2) is not available using specific swimlane object, then it still get me that property from different swimlane feedcard as it identify its visible. If it does use root element reference to find specific feed card property, then it should not have this problem.

Following is the structure:
public class SwimlanesPage extends MobileAbstractPage implements IMobileUtils {
    
    @ExtendedFindBy(iosClassChain = "**/XCUIElementTypeCell[`name == \"ListTableCell\" OR name == \"DetailCastCell\"`]")
    protected List<Swimlane> swimlanes;
}
public class Swimlane extends AbstractUIObject implements IMobileUtils {

    @ExtendedFindBy(iosClassChain = "**/XCUIElementTypeCell[`name == \"PosterCollectionCell\" OR name == \"RoleCell\" OR name ==\"PosterCell\"`]")
    protected List<FeedCard> feedCards;
}
public class FeedCard extends AbstractUIObject implements IMobileUtils {

    @ExtendedFindBy(iosClassChain = "**/XCUIElementTypeImage[`name == \"PosterCollectionImgPoster\" OR name == \"PosterImageView\" OR name == \"RoleImageView\"`]")
    private ExtendedWebElement image;

    @ExtendedFindBy(iosClassChain = "**/XCUIElementTypeStaticText[`name == \"Text1\" OR name == \"RoleLabelText1\"`]")
    private ExtendedWebElement text1;

    @ExtendedFindBy(iosClassChain = "**/XCUIElementTypeStaticText[`name == \"Text2\" OR name == \"RoleLabelText2\"`]")
    private ExtendedWebElement text2;
}

**Comment by DimaGrinevich** [comment]
Now Carina first tries to find an element using SearchContext. If element wasn't found Carina will try to search it without searchContext. If you wan to search elements only with searchContext in every Carina method like isVisible(), isClickable() etc. set true at the end of method)

element.isVisible(long timeout, boolean bySearchContext)
element.isVisible(5, true) - search element only with searchContext
element.isVisible(5, false) - search element with searchContext, if not found search without searchContext

**Comment by tvtester** [comment]
I updated <carina-core_version>7.2.10</carina-core_version> in my pom.xml, but still not seeing above changes in the ExtendedWebElement.java class:
<img width="973" alt="Screen Shot 2021-09-09 at 5 11 46 PM" src="https://user-images.githubusercontent.com/47870905/132763480-349aebca-4136-40a3-9a22-e51451328f1f.png">

**Comment by DimaGrinevich** [comment]
Yep, you are right. Created new **7.2.10-searchFix-SNAPSHOT** build that involves this fix.

**Comment by vdelendik** [comment]
@tvtester, is it possible to test on your end using `7.3.15-SNAPSHOT`. We refactored a lot this logic inside making it much more clear based on 1st review :)

**Comment by tvtester** [comment]
I have tried the Last snapshot: 7.2.12-elementSearch-SNAPSHOT
However, still object is not identified, and timeout exception is thrown below.

2021-11-27 14:01:05 ExtendedWebElement [main-1] [DEBUG] waitUntil: starting... timeout: 1
2021-11-27 14:01:09 ExtendedWebElement [main-1] [DEBUG] waitUntil: TimeoutException: Expected condition failed: waiting for visibility of [[[[io.appium.java_client.ios.IOSDriver, Capabilities: {app=com.telus.opusprod, automationName=XCUITest, browserName=, carinaTestRunId=67f73840-9443-4490-ae61-e1261ef8e56a, databaseEnabled=false, deviceName=Apple TV TSOQA5, forceAppLaunch=true, javascriptEnabled=true, locationContextEnabled=false, networkConnectionEnabled=false, newCommandTimeout=30000, noReset=true, platform=MAC, platformName=tvOS, platformVersion=13.4.8, settingsApp=com.apple.TVSettings, shouldTerminateApp=true, takesScreenshot=true, udid=e791c376d010360adc7df0e0bdb2759f021cbe98, waitForAppScript=true, webStorageEnabled=false, xcodeOrgId=NUN484DX7A, xcodeSigningId=iPhone Developer}] -> xpath: //XCUIElementTypeScrollView/XCUIElementTypeCollectionView[@name="FeedList"]/XCUIElementTypeCell[contains(@name,"PosterCell") and @focused="true"]]] -> xpath: .//XCUIElementTypeImage[@name="OutOfHome"]] (tried for 1 second(s) with 10 milliseconds interval)
Build info: version: '3.141.59', revision: 'e82be7d358', time: '2018-11-14T08:17:03'
System info: host: 'Haos-MacBook-Pro.local', ip: '127.0.0.1', os.name: 'Mac OS X', os.arch: 'x86_64', os.version: '10.16', java.version: '15.0.2'
Driver info: org.openqa.selenium.support.events.EventFiringWebDriver
Capabilities {app: com.telus.opusprod, automationName: XCUITest, browserName: , carinaTestRunId: 67f73840-9443-4490-ae61-e12..., databaseEnabled: false, deviceName: Apple TV TSOQA5, forceAppLaunch: true, javascriptEnabled: true, locationContextEnabled: false, networkConnectionEnabled: false, newCommandTimeout: 30000, noReset: true, platform: MAC, platformName: tvOS, platformVersion: 13.4.8, settingsApp: com.apple.TVSettings, shouldTerminateApp: true, takesScreenshot: true, udid: e791c376d010360adc7df0e0bdb..., waitForAppScript: true, webStorageEnabled: false, xcodeOrgId: NUN484DX7A, xcodeSigningId: iPhone Developer}
Session ID: f60ef9cb-8893-4a85-8687-ebb1b1d75e49
2021-11-27 14:01:09 ExtendedWebElement [main-1] [DEBUG] waitUntil: starting... timeout: 20
2021-11-27 14:01:36 ExtendedWebElement [main-1] [DEBUG] waitUntil: TimeoutException: Expected condition failed: waiting for visibility of [[[[io.appium.java_client.ios.IOSDriver, Capabilities: {app=com.telus.opusprod, automationName=XCUITest, browserName=, carinaTestRunId=67f73840-9443-4490-ae61-e1261ef8e56a, databaseEnabled=false, deviceName=Apple TV TSOQA5, forceAppLaunch=true, javascriptEnabled=true, locationContextEnabled=false, networkConnectionEnabled=false, newCommandTimeout=30000, noReset=true, platform=MAC, platformName=tvOS, platformVersion=13.4.8, settingsApp=com.apple.TVSettings, shouldTerminateApp=true, takesScreenshot=true, udid=e791c376d010360adc7df0e0bdb2759f021cbe98, waitForAppScript=true, webStorageEnabled=false, xcodeOrgId=NUN484DX7A, xcodeSigningId=iPhone Developer}] -> xpath: //XCUIElementTypeScrollView/XCUIElementTypeCollectionView[@name="FeedList"]/XCUIElementTypeCell[contains(@name,"PosterCell") and @focused="true"]]] -> xpath: .//XCUIElementTypeImage[@name="OutOfHome"]] (tried for 20 second(s) with 10 milliseconds interval)
Build info: version: '3.141.59', revision: 'e82be7d358', time: '2018-11-14T08:17:03'
System info: host: 'Haos-MacBook-Pro.local', ip: '127.0.0.1', os.name: 'Mac OS X', os.arch: 'x86_64', os.version: '10.16', java.version: '15.0.2'
Driver info: org.openqa.selenium.support.events.EventFiringWebDriver
Capabilities {app: com.telus.opusprod, automationName: XCUITest, browserName: , carinaTestRunId: 67f73840-9443-4490-ae61-e12..., databaseEnabled: false, deviceName: Apple TV TSOQA5, forceAppLaunch: true, javascriptEnabled: true, locationContextEnabled: false, networkConnectionEnabled: false, newCommandTimeout: 30000, noReset: true, platform: MAC, platformName: tvOS, platformVersion: 13.4.8, settingsApp: com.apple.TVSettings, shouldTerminateApp: true, takesScreenshot: true, udid: e791c376d010360adc7df0e0bdb..., waitForAppScript: true, webStorageEnabled: false, xcodeOrgId: NUN484DX7A, xcodeSigningId: iPhone Developer}
Session ID: f60ef9cb-8893-4a85-8687-ebb1b1d75e49


I tried to use 7.3.15-SNAPSHOT but script fails right away by complaining activeApp(bundleID) unknown command. Removed activeApp(bundleID), and tried then, right away it skip execution and complaining search context is null.
